nexus_keycode/protocols/channel_origin_commands.py

```diff
@@ -0,0 +1,458 @@
+        'look up' the accessory Nexus ID to validate the message.
+
+        Practically, we can do this by allowing the origin manager to 'ask'
+        for the ID of all linked accessories (optionally those matching the
+        truncated ID), and attempt to compute the MAC using each of those. If
+        there is no match, the message is invalid.
+
+        The expanded 'body' consists of two parts - the Nexus 'authority'
+        ID (first 2 bytes of Nexus ID), and the Nexus 'device' ID (last 4 bytes
+        of Nexus ID).
+        """
+
+        # Requires 16-byte symmetric Nexus keys
+        assert len(controller_sym_key) == 16
+
+        # note that these are not 'transmitted' body digits, but it is
+        # assumed that the receiver will 'expand' the message (from the
+        # transmitted, truncated accessory ID) to then find any linked
+        # accessory with a matching device ID, and pull in the full values
+        # to use to generate this MAC.
+
+        # vendor / authority ID is upper 2 bytes of the full 'Nexus ID'
+        nexus_authority_id = (accessory_nexus_id & 0xFFFF00000000) >> 32
+        nexus_device_id = accessory_nexus_id & 0xFFFFFFFF
+
+        packed_target_inputs = bitstring.pack(
+            [
+                "uintle:32=controller_command_count",
+                "uintle:8=origin_command_type_code",  # '2' or '3'
+                "uintle:16=nexus_authority_id",
+                "uintle:32=nexus_device_id",
+            ],
+            controller_command_count=controller_command_count,
+            origin_command_type_code=type_.value,
+            nexus_authority_id=nexus_authority_id,
+            nexus_device_id=nexus_device_id
+        ).bytes
+
+        assert len(packed_target_inputs) == 11
+        auth = siphash.SipHash_2_4(
+            controller_sym_key,
+            packed_target_inputs
+        )
+
+        return cls(
+            type_=type_,
+            accessory_nexus_id=accessory_nexus_id,
+            auth=auth
+        )
+
+    @classmethod
+    def unlink_specific_accessory(
+            cls,
+            accessory_nexus_id,
+            controller_command_count,
+            controller_sym_key
+    ):
+        return cls._specific_accessory_builder(
+            OriginCommandType.UNLINK_ACCESSORY,
+            accessory_nexus_id,
+            controller_command_count,
+            controller_sym_key
+        )
+
+    @classmethod
+    def unlock_specific_accessory(
+            cls,
+            accessory_nexus_id,
+            controller_command_count,
+            controller_sym_key
+    ):
+        return cls._specific_accessory_builder(
+            OriginCommandType.UNLOCK_ACCESSORY,
+            accessory_nexus_id,
+            controller_command_count,
+            controller_sym_key
+        )
+
+
+class LinkCommandToken(ChannelOriginCommandToken):
+
+    _origin_command_type = OriginCommandType.LINK_ACCESSORY_MODE_3
```

Comment:
Since its not a "generic origin command", I don't think we hit this line:

https://github.com/angaza/nexus-python/pull/14/files#diff-17923aa0f4a021979d134f389ed9099b18551de149038a366d91c11ef39d2329R232

I'll remove it.

Reply:
Agree (it's still there).